bitcoin/src/blockdata/script.rs

```diff
@@ -1222,6 +1232,18 @@ mod test {
         script = script.push_opcode(OP_CHECKSIG); comp.push(0xACu8); assert_eq!(&script[..], &comp[..]);
     }
 
+    #[test]
+    fn test_p2pk_public_key() {
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_public_key().unwrap();
+        assert_eq!(actual, key);
+
+        let invalid_script = Script::builder().push_key(&key).into_script();
+        assert_eq!(invalid_script.p2pk_public_key(), None);
+    }
```

Comment:
Consider adding a test for malformed key.

Reply:
Sure